tests/suite/graceful_recovery_test.go

```diff
@@ -0,0 +1,334 @@
+package suite
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/batch/v1"
+	coordination "k8s.io/api/coordination/v1"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+const (
+	teaURL             = "https://cafe.example.com/tea"
+	coffeeURL          = "http://cafe.example.com/coffee"
+	nginxContainerName = "nginx"
+	ngfContainerName   = "nginx-gateway"
+)
+
+var _ = Describe("Graceful Recovery test", Ordered, Label("nfr", "graceful-recovery"), func() {
+	files := []string{
```

Comment:
When should a variable such as `files` or `ns` be declared in this describe, vs as a constant? 

I know that with the helper functions that I've defined, they could either use variables as a constant, or if they are in the describe, they would need to accept them as parameters from when they are called inside the describe.

I've added a function called `runTest` which accepts `files` and `ns` as arguments, but I wasn't sure if these variables should be constants since this helper function would most likely stay local to this file.

Reply:
A constant will be defined for the package. Functions are also available to be used anywhere in the package. Exported functions are for use outside of the package by other packages.

Files don't restrict the scope of functions or variables to that file, everything is at least package-wide (unless the variables are within a function, of course).